pkg/sql/plan2/function/operator/cast.go

```diff
@@ -755,6 +755,35 @@ func CastSameType2[T types.Date | types.Datetime | types.Timestamp](lv, rv *vect
 	return vec, nil
 }
 
+//It's an error to use CastLeftToRight when convert float to integer,So this is a special fix
+func CastFloatToInteger[T1 constraints.Float, T2 constraints.Integer](lv, rv *vector.Vector, proc *process.Process) (*vector.Vector, error) {
```

Comment:
What is the error?   What is the diff between your version and the CastLeftToRight?  Can you add that in comment?

Reply:
I'll give the comments in the codes, the error is that: int the old code, if you go through the CastLeftToRight , like 1.5 will be 1, and it should be 2, but if I modify the CastLeftToRight straight, it will cause other things ,So as to avoid that, I give another func CastFloatToInteger